packages/data-core/src/store/store.ts

```diff
@@ -13,31 +13,44 @@ export interface Store {
 
 export const store: Store = {}
 
+/**
+ * Initializes the store with a ResourceStore for each schema.
+ */
 export function createStore(schemas: string[]): Store {
-  schemas.forEach((name) => {
-    store[name] = {
+  for (let i = 0; i < schemas.length; i++) {
+    store[schemas[i]] = {
       data: {},
       subscribers: [],
     }
-  })
+  }
 
   return store
 }
 
+/**
+ * Returns the ResourceStore for a given schema.
+ */
 export function getStore(schema: string): ResourceStore {
   return store[schema]
 }
 
+/**
+ * Inserts data into the store and notifies subscribers.
+ */
 export function insert(schema: string, data: Resource[]): void {
   store[schema].data = {
     ...store[schema].data,
     ...keyResourcesById(data),
   }
 
   const records = data.map(convertResourceToRecord)
+
   store[schema].subscribers.forEach((subscriber) => subscriber(records))
```

Comment:
i left this one as `forEach` as I thought it was more readable than:
```
for (let i = 0; i < store[schema].subscribers.length; i++) 
    store[schema].subscribers[i](records)
```
thoughts?

Reply:
Only use an index counter if you actually need the index. for this I would use
```
for (const subscriber of store[schema].subscribers)  {
```